Movies page: Api route + styling

rl-86 opened the pull request:
### API fetch
I have updated MovieList and MovieDetails components to fetch movie data from our API route, /api/movies.
_____________________________

### Styling
I have added basic MUI styling to the Movie overview page and movie details pages.

______________________________________________________

Edit: Made the pages more responsive.

**Movies page desktop**
![Skärmbild 2024-05-20 221447](https://github.com/JrSchoolAccount/Kino-site-nextjs/assets/144668940/c5f78fc5-aa18-4829-817a-070ec2b9bf19)

**Movie details desktop**
![Skärmbild 2024-05-20 204643](https://github.com/JrSchoolAccount/Kino-site-nextjs/assets/144668940/21198613-ddc6-4c58-924d-ea730a82978b)


**Movies page mobile**
![Skärmbild 2024-05-20 221518](https://github.com/JrSchoolAccount/Kino-site-nextjs/assets/144668940/4921381b-2997-4b16-a7b9-cba93a0a1797)


**Movie details mobile**
![Skärmbild 2024-05-20 204730](https://github.com/JrSchoolAccount/Kino-site-nextjs/assets/144668940/3b826849-5cde-4a96-ae2e-d945194d77cf)

NelsonTheBrave reviewed (approved):
Good job Richard, it looks really nice, and well done with making it responsive as well :D 
There are some very small things that could be done, but I will approve it.
I'm thinking of the movie plot text that ends with "...", it would be nice to avoid that. Also there is a särskrivning in IMDB-betyg, other than that I think it's good :)

simonpizevski reviewed (approved):
I really like how it looks now, good job with the styling and the fetching of movies.

rl-86 commented:
> Good job Richard, it looks really nice, and well done with making it responsive as well :D There are some very small things that could be done, but I will approve it. I'm thinking of the movie plot text that ends with "...", it would be nice to avoid that. Also there is a särskrivning in IMDB-betyg, other than that I think it's good :)

Thanks!
I have addressed these points and made the pages a little more responsive as well :)
